aws/connection_config.go

```diff
@@ -5,59 +5,28 @@ import (
 	"strings"
 
 	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
-	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/schema"
 )
 
 type awsConfig struct {
-	Regions               []string `cty:"regions"`
-	DefaultRegion         *string  `cty:"default_region"`
-	Profile               *string  `cty:"profile"`
-	AccessKey             *string  `cty:"access_key"`
-	SecretKey             *string  `cty:"secret_key"`
-	SessionToken          *string  `cty:"session_token"`
-	MaxErrorRetryAttempts *int     `cty:"max_error_retry_attempts"`
-	MinErrorRetryDelay    *int     `cty:"min_error_retry_delay"`
-	IgnoreErrorCodes      []string `cty:"ignore_error_codes"`
-	EndpointUrl           *string  `cty:"endpoint_url"`
-	S3ForcePathStyle      *bool    `cty:"s3_force_path_style"`
+	Regions               []string          `hcl:"regions,optional"`
+	DefaultRegion         *string           `hcl:"default_region,optional"`
+	Profile               *string           `hcl:"profile,optional"`
+	AssumeRole            *assumeRoleConfig `hcl:"assume_role,optional"`
+	AccessKey             *string           `hcl:"access_key,optional"`
+	SecretKey             *string           `hcl:"secret_key,optional"`
+	SessionToken          *string           `hcl:"session_token,optional"`
+	MaxErrorRetryAttempts *int              `hcl:"max_error_retry_attempts,optional"`
+	MinErrorRetryDelay    *int              `hcl:"min_error_retry_delay,optional"`
+	IgnoreErrorCodes      []string          `hcl:"ignore_error_codes,optional"`
+	EndpointUrl           *string           `hcl:"endpoint_url,optional"`
+	S3ForcePathStyle      *bool             `hcl:"s3_force_path_style,optional"`
 }
 
-var ConfigSchema = map[string]*schema.Attribute{
```

Comment:
@brittandeyoung Why did you remove this along with `Schema:      ConfigSchema,` in plugin.go?

Reply:
@cbruno10 This was done in order to support the map for `assume_role` as the schema does not support map objects. The idea was to use tagging to dynamically pull in the settings. 

I had this working with root level configurations instead of a map and utilizing the defined schema. Would this pull request be better to be reverted to using root level configurations instead of a map configuration?

Reply:
Thanks for highlighting this; I have raised a Steampipe SDK [issue](https://github.com/turbot/steampipe-plugin-sdk/issues/691) to track.

Reply:
@bigdatasourav Great! This getting added to the SDK will likely enable me to complete this PR with working code.